dist/general_test.go

```diff
@@ -157,3 +160,45 @@ func parametersEqual(p1, p2 []Parameter, tol float64) bool {
 	}
 	return true
 }
+
+type derivParamTester interface {
+	LogProb(x float64) float64
+	DLogProbDParam(x float64, deriv []float64)
+	Quantile(p float64) float64
+	NumParameters() int
+	parameters([]Parameter) []Parameter
+	setParameters([]Parameter)
+}
+
+func testDerivParam(t *testing.T, d derivParamTester) {
+	// Tests that the derivative matches for a number of different quantiles
+	// along the distribution.
+	nTest := 10
+	quantiles := make([]float64, nTest)
+	floats.Span(quantiles, 0.1, 0.9)
+
+	deriv := make([]float64, d.NumParameters())
+	fdDeriv := make([]float64, d.NumParameters())
+
+	initParams := d.parameters(nil)
+	init := make([]float64, d.NumParameters())
+	for i, v := range initParams {
+		init[i] = v.Value
+	}
+	for _, v := range quantiles {
+		x := d.Quantile(v)
+		d.DLogProbDParam(x, deriv)
```

Comment:
I think this line reuses parameters set in the previous for iteration inside the closure. If `d.setParameters(initParams)` is added before it, the tolerance below can be tightened to 1e-9.

Other than this, it LGTM.